test/suite/test_compact10.py

```diff
@@ -0,0 +1,212 @@
+#!/usr/bin/env python
+#
+# Public Domain 2014-present MongoDB, Inc.
+# Public Domain 2008-2014 WiredTiger, Inc.
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import os, time, wttest
+from wiredtiger import stat
+from wtbackup import backup_base
+
+# test_compact10.py
+# Verify backups can be taken while background compaction is running.
+class test_compact10(backup_base):
+    backup_dir_1 = "BACKUP_1"
+    backup_dir_2 = "BACKUP_2"
+    conn_config = 'cache_size=100MB,statistics=(all)'
+    create_params = 'key_format=i,value_format=S,allocation_size=4KB,leaf_page_max=32KB'
+    uri_prefix = 'table:test_compact10'
+
+    num_tables = 5
+    table_numkv = 100 * 1000
+    value_size = 1024 # The value should be small enough so that we don't create overflow pages.
+
+    def delete_range(self, uri, num_keys):
+        c = self.session.open_cursor(uri, None)
+        for i in range(num_keys):
+            c.set_key(i)
+            c.remove()
+        c.close()
+
+    def get_bg_compaction_running(self):
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        compact_running = stat_cursor[stat.conn.background_compact_running][2]
+        stat_cursor.close()
+        return compact_running
+
+    def get_bytes_recovered(self):
+        stat_cursor = self.session.open_cursor('statistics:', None, None)
+        bytes_recovered = stat_cursor[stat.conn.background_compact_bytes_recovered][2]
+        stat_cursor.close()
+        return bytes_recovered
+
+    def get_files_compacted(self, uris):
+        files_compacted = 0
+        for uri in uris:
+            if self.get_pages_rewritten(uri) > 0:
+                files_compacted += 1
+        return files_compacted
+
+    def get_pages_rewritten(self, uri):
```

Comment:
I wonder if its cleaner if we use a get_stats function here, e.g.
```
    # Verify statistics.
    def get_stat(self, stat):
        stat_cursor = self.session.open_cursor('statistics:', None, None)
        val = stat_cursor[stat][2]
        stat_cursor.close()
        return val
        
```

Reply:
The long-term solution would be to have a `compactbase.py` to define everything that is common to all the compact test. I have made changes in 7f5919a791fdd198427061499510607eae0c6238 for now.